compiler/rustc_traits/src/codegen.rs

```diff
@@ -72,6 +72,10 @@ pub fn codegen_select_candidate<'tcx>(
 
     let impl_source = infcx.resolve_vars_if_possible(impl_source);
     let impl_source = infcx.tcx.erase_regions(impl_source);
+    if impl_source.has_infer() {
+        infcx.tcx.dcx().has_errors().unwrap();
+        return Err(CodegenObligationError::FulfillmentError);
+    }
```

Comment:
Unused lifetimes on an impl get replaced with inference vars, but never resolved, causing the return value of a query to contain inference vars, which is

1. bad
2. ICEs in stable hashing (good!)

Reply:
Can we put this comment as a comment in the code?